src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+        }
+        return item;
+
+    } else if (item == ITEM_KEY_SMALL) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_OCARINA) || (item == ITEM_BOMBCHU) || (item == ITEM_HOOKSHOT) || (item == ITEM_LENS) ||
+               (item == ITEM_SWORD_GREAT_FAIRY) || (item == ITEM_PICTO_BOX)) {
+        if (INV_CONTENT(item) == ITEM_NONE) {
+            return ITEM_NONE;
+        }
+        return INV_CONTENT(item);
+
+    } else if ((item >= ITEM_BOMBS_5) && (item == ITEM_BOMBS_30)) {
+        //! @bug: Should be a range check: (item <= ITEM_BOMBS_30)
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item >= ITEM_BOMBCHUS_20) && (item <= ITEM_BOMBCHUS_5)) {
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item == ITEM_QUIVER_30) || (item == ITEM_BOW)) {
+        if (CUR_UPG_VALUE(UPG_QUIVER) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item == ITEM_QUIVER_40) || (item == ITEM_QUIVER_50)) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_BOMB_BAG_20) || (item == ITEM_BOMB)) {
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            return ITEM_NONE;
+        }
+        return 0;
+
+    } else if ((item >= ITEM_STICK_UPGRADE_20) && (item <= ITEM_NUT_UPGRADE_40)) {
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_BOMB_BAG_30) && (item <= ITEM_WALLET_GIANT)) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BEAN) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_POWDER_KEG) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_HEART_PIECE_2) || (item == ITEM_HEART_PIECE)) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_HEART_CONTAINER) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_HEART) {
+        return ITEM_HEART;
+
+    } else if ((item == ITEM_MAGIC_SMALL) || (item == ITEM_MAGIC_LARGE)) {
+        if (!(gSaveContext.save.weekEventReg[12] & 0x80)) {
+            return ITEM_NONE;
+        }
+        return item;
+
+    } else if ((item >= ITEM_RUPEE_GREEN) && (item <= ITEM_RUPEE_HUGE)) {
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_REMAINS_ODOLWA) && (item <= ITEM_REMAINS_TWINMOLD)) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_LONGSHOT) {
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BOTTLE) {
+        return ITEM_NONE;
+
+    } else if ((item == ITEM_MILK_BOTTLE) || (item == ITEM_POE) || (item == ITEM_GOLD_DUST) || (item == ITEM_CHATEAU) ||
+               (item == ITEM_HYLIAN_LOACH)) {
+        return ITEM_NONE;
+
+    } else if (((item >= ITEM_POTION_RED) && (item <= ITEM_OBABA_DRINK)) || (item == ITEM_CHATEAU_2) ||
+               (item == ITEM_MILK) || (item == ITEM_GOLD_DUST_2) || (item == ITEM_HYLIAN_LOACH_2) ||
+               (item == ITEM_SEA_HORSE_CAUGHT)) {
+        temp = SLOT(item);
```

Comment:
```suggestion
        slot = SLOT(item);
```
Seems to store a slot?